proc/proc_darwin.go

```diff
@@ -292,11 +323,26 @@ func (dbp *Process) findExecutable(path string) (*macho.File, error) {
 }
 
 func (dbp *Process) trapWait(pid int) (*Thread, error) {
+	trapWaitLoop:
 	for {
-		port := C.mach_port_wait(dbp.os.portSet, C.int(0))
+		task := dbp.os.task
+		port := C.mach_port_wait(dbp.os.portSet, &task, C.int(0))
 
 		switch port {
 		case dbp.os.notificationPort:
+			// on macOS >= 10.12.1 the task_t changes after an execve, we could
+			// receive the notification for the death of the pre-execve task_t,
+			// this could also happen *before* we are notified that our task_t has
+			// changed.
+			if dbp.os.task != task {
+				fmt.Printf("ignoring death of %d (%d)\n", task, dbp.os.task)
+				continue trapWaitLoop
+			}
+			if pidtask := C.get_task_for_pid(C.int(dbp.Pid)); pidtask != 0 && dbp.os.task != pidtask {
+				fmt.Printf("ignoring death of %d (%d) - 2\n", pidtask, dbp.os.task)
```

Comment:
nit: remove this print lines, or use proper logging which we can enable / disable via log flag.